### PR TITLE
peering: update unit tests to not reuse peering token

### PR DESCRIPTION
**Pull request by ndhanushkodi**
Changes proposed in this PR:
- Removing some code that edits the peering token after generating to add server addresses(we don't need to do this anymore because the token has the correct server addresses in it)
- Doesn't reuse the same peering token twice-- if a peering token has already been used, we need to generate a new one if we want to update the peering during a reconcile loop.

How I've tested this PR:
- Ran unit tests with 1.13

How I expect reviewers to test this PR:
- 👀 

Checklist:
- [ ] Tests added
- [ ] CHANGELOG entry added 
  > HashiCorp engineers only, community PRs should not add a changelog entry.
  > Entries should use present tense (e.g. Add support for...)

### CODE REVIEW

File: .github/workflows/test.yml
Comment (ndhanushkodi) [inline .github/workflows/test.yml:149]: This is based of of 1.13.x branch of consul

**Review by thisisnotashwin — APPROVED** [review]

**Review by ishustava — APPROVED** [review]

File: .github/workflows/test.yml
Comment (ishustava) [reply .github/workflows/test.yml:149]: Should we wait for core release before merging?

Comment (ndhanushkodi) [reply .github/workflows/test.yml:149]: ya that works or I can make an explicit revert change to switch back to releases.hashicorp.com right after core release-- i think if it takes them past the early afternoon, then I'll merge this for today and make the explicit revert change! that way we can have our unit tests running against 1.13 on main as we're merging other k8s PRs too

Comment (ndhanushkodi) [reply .github/workflows/test.yml:149]: theyve released! So i'll update this PR

**Pull request merged**
